pkg/services/sqlstore/searchstore/builder.go

```diff
@@ -0,0 +1,108 @@
+package searchstore
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+	"strings"
+)
+
+type Builder struct {
+	Filters []interface{}
+	Dialect migrator.Dialect
+
+	params []interface{}
+	sql    bytes.Buffer
+}
+
+// ToSql builds the sql and returns it as a string, together with the params.
+func (b *Builder) ToSql(limit, page int64) (string, []interface{}) {
+	b.params = make([]interface{}, 0)
+	b.sql = bytes.Buffer{}
+
+	b.buildSelect()
+
+	b.sql.WriteString("( ")
+	b.applyFilters()
+
+	b.sql.WriteString(b.Dialect.LimitOffset(limit, (page-1)*limit) + `) as ids
+		INNER JOIN dashboard on ids.id = dashboard.id
+	`)
+
+	b.sql.WriteString(`
+		LEFT OUTER JOIN dashboard folder on folder.id = dashboard.folder_id
```

Comment:
Is `LEFT OUTER JOIN dashboard folder` correct syntax?I would think there should just be the name of a single table to join?
```suggestion
		LEFT OUTER JOIN dashboard folder ON folder.id = dashboard.folder_id
```

Reply:
I copied it over from the `search_builder.go`. I was a bit confused by this as well, I can remove it and see if it makes a difference.

Reply:
Thanks @sakjur :)

Reply:
I looked at this and consulted @xlson. We _don't_ have a folder's table, it's a rename for `dashboard` 🙂 I clarified it by adding `AS`